doc/nrf/migration/migration_guide_1.x_to_2.x.rst

```diff
@@ -0,0 +1,525 @@
+This option is now enabled by default to allow a smooth transition.
+In order to facilitate the migration to the new include prefix, a script to automate the process is also provided in :file:`scripts/utils/migrate_includes.py` (in Zephyr).
+
+Changes in PWM API
+******************
+
+Zephyr v3.x.x introduces changes in the PWM API that require modifying the board definitions.
+Old board definitions will cause a compilation error, and calling the old API functions will result in warnings stating that these functions are deprecated.
+
+Required action:
+   * ``pwms`` properties in devicetree nodes must be extended with two more cells (with period and flags) and now they need to specify PWM channels, not pin numbers.
+   * Calls to the deprecated ``pwm_pin_set_cycles`` function must be replaced with calls to the :c:func:`pwm_set_cycles` function.
+   * Calls to the deprecated ``pwm_pin_set_usec`` and ``pwm_pin_set_nsec`` functions must be replaced with calls to the :c:func:`pwm_set` function with the period and pulse values wrapped in the :c:macro:`PWM_USEC` macro or the :c:macro:`PWM_NSEC` macro, respectively.
+
+Note that the :c:func:`pwm_set` and :c:func:`pwm_set_cycles` functions take a PWM channel as a parameter, not a pin number as the deprecated functions did.
+Also, the ``flags`` parameter is now supported, so either the :c:macro:`PWM_POLARITY_INVERTED` or :c:macro:`PWM_POLARITY_NORMAL` flag must be provided in each call.
+
+Wherever possible, it is recommended to use the newly introduced :c:macro:`PWM_DT_SPEC_GET` macro (or another suitable one from its family) to obtain PWM information from devicetree, and then use the :c:func:`pwm_set_dt` or :c:func:`pwm_set_pulse_dt` function instead of :c:func:`pwm_set`.
+
+For example, for PWM channels defined as follows:
+
+.. code-block:: devicetree
+
+    pwm0_default: pwm0_default {
+        group1 {
+            psels = <NRF_PSEL(PWM_OUT0, 0, 11)>;
+            nordic,invert;
+        };
+        group2 {
+            psels = <NRF_PSEL(PWM_OUT3, 1, 5)>;
+        };
+    };
+
+    pwm0_sleep: pwm0_sleep {
+        group1 {
+            psels = <NRF_PSEL(PWM_OUT0, 0, 11)>,
+                    <NRF_PSEL(PWM_OUT3, 1, 5)>;
+            low-power-enable;
+        };
+    };
+
+    &pwm0 {
+        status = "okay";
+        pinctrl-0 = <&pwm0_default>;
+        pinctrl-1 = <&pwm0_sleep>;
+        pinctrl-names = "default", "sleep";
+    };
+
+You must update the PWM LED definitions that use those channels:
+
+.. code-block:: devicetree
+
+    /* old definitions that will no longer work */
+    pwm_led0: pwm_led_0 {
+        pwms = <&pwm0 11>;
+    };
+    pwm_led1: pwm_led_1 {
+        pwms = <&pwm0 37>;
+    };
+
+.. note::
+   The period lengths, set here arbitrarily to commonly used value of 20 ms, are provided as default ones.
+   They can be overridden in the actual PWM API calls if needed.
```

Comment:
I think that it would be better to place this note after the code snippet that it refers to, i.e. after the one with updated definitions. In the current form it is a bit confusing, at least to me.

Reply:
moved